hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/cli/GenericCli.java

```diff
@@ -63,25 +61,11 @@ public GenericCli(Class<?> type) {
     });
 
     if (type != null) {
-      addSubcommands(getCmd(), type);
+      SubcommandWithParent.addSubcommands(getCmd(), type);
     }
     ExtensibleParentCommand.addSubcommands(cmd);
```

Comment:
Thanks @nandakumar131 for reviewing this.

This PR moves all known subcommands (in this repo) to the fine-grained interface.  So we could simply drop `SubcommandWithParent` here.  Instead, I kept the existing logic as deprecated to not break behavior for subcommands we don't know about (used privately).

On another look, `SubcommandWithParent` support was dropped for `ozone admin` in HDDS-11831 and for `ozone sh` in HDDS-11822.

I kind of plan to make another change where `type` is simply passed from `GenericCli` constructor as `getClass()`, instead of requiring all subclasses to pass their own `.class`.  That would restore behavior for `ozone admin` and `ozone sh`, too.

What do you think?

Reply:
Removing the `type` parameter may not be ideal, as some subclasses are not be intended to be extensible (mostly server-side component starter commands, like `OzoneManagerStarter` or `HddsDatanodeService`).

Reply:
If we can move all the commands to use `ExtensibleParentCommand` logic which is fine-grained interface for all the sub-commands, then we can remove the `type`. To make this we have to change the sub-command loading logic in `OzoneShell`, which makes sense. We will not have multiple ways in which we load the sub-commands, everything will go through the same code flow which will be easier to understand.